test/modules/TestResultCoordinator/Reports/CountingReportGenerator.cs

```diff
@@ -112,19 +126,28 @@ public async Task<ITestResultReport> CreateReportAsync()
             }
 
             // Check duplicates at the end of actual results
-            while (hasActualResult && this.TestResultComparer.Matches(lastLoadedResult, this.ActualTestResults.Current))
+            while (hasActualResult && this.TestResultComparer.Matches(lastLoadedActualResult, this.ActualTestResults.Current))
```

Comment:
Please consider the case
Actual: 1 -> 2 -> 3 -> 4
Expect: 1-> 2 -> 3

This should be considered a fail. With the current logic it would be a pass.

Reply:
I disagree because this while loop makes sure the lastLoadedActualResult matches the current.